test-integration/helpers/test-element.coffee

```diff
@@ -48,26 +51,37 @@ class TestItemHelper
 
   isPresent: (args...) =>
     locator = @getLocator(args...)
-    @test.driver.isElementPresent(locator).then (isPresent) ->
-      isPresent
+    @test.driver.isElementPresent(locator)
+
+  isDisplayed: (args...) =>
+    @isPresent(args...).then (isPresent) =>
+      if isPresent
+        locator = @getLocator(args...)
+        el = @findElement(args...)
+        el.isDisplayed()
+      else
+        false
 
   # Helper for the common case of `get(...).click()`.
   # Plus, it allows a place to add logging since this is one of the most
   # common places for Selenium to time out (trying to click on an element)
   click: (args...) =>
     locator = @getLocator(args...)
+    # Scroll to the element so it is visible before clicking (this assumes `position: fixed` is overridden for all element)
+    # @isDisplayed(args...).then (isDisplayed) =>
+    #   unless isDisplayed
+    #     el = @findElement(args...)
+    #     @test.utils.windowPosition.scrollTo(el)
     @test.utils.verboseWrap "Clicking #{JSON.stringify(locator)}", =>
-      @get(args...).click()
+      el = @get(args...)
+      @test.utils.windowPosition.scrollTo(el)
+      el.click()
 
-# Using defined properties for access eliminates the possibility
-# of accidental assignment
-Object.defineProperties TestItemHelper.prototype,
-  test:
-    get: -> @_test
-  locator:
-    get: -> @_locator
-  element:
-    get: -> @get()
+  waitClick: (args...) =>
+    locator = @getLocator(args...)
+    @test.utils.verbose "Waiting for #{JSON.stringify(locator)}"
+    @test.utils.wait.for(locator)
+    @click(args...)
 
```

Comment:
Yeah, I think they might be the same. I was wondering if the semantics for `click` should be that it does not wait. That way, if you are expecting an element to be present, Selenium will error immediately instead of waiting for the element to "eventually appear" and then time out. Also, that would help ensure that the test waited for the page to be loaded first. Thoughts?